addons/account/models/sequence_mixin.py

```diff
@@ -239,7 +240,12 @@ def _set_next_sequence(self):
         :param field_name: the field that contains the sequence.
         """
         self.ensure_one()
-        last_sequence = self._get_last_sequence()
+        # if (?P<no_reset>) is on the regex then do NOT reset the sequence on change of
+        # period
+        # check the monthly regex since it is the first one being checked on
+        # _deduce_sequence_number_reset
+        no_reset = "(?P<no_reset>)" in self._sequence_monthly_regex
+        last_sequence = self._get_last_sequence(relaxed=no_reset)
```

Comment:
Shouldn't we do this below?
```python
        if new:
            format_values['seq'] = 0
        if new or no_reset:
            format_values['year'] = self[self._sequence_date_field].year % (10 ** format_values['year_length'])
            format_values['month'] = self[self._sequence_date_field].month
```

Reply:
You're right. I added a test that catches this oversight.